src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments/Queries/GetTrainingCourses/GetTrainingCoursesQueryHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.ProviderCommitments.Domain_Models.ApprenticeshipCourse;
+using SFA.DAS.ProviderCommitments.Extensions;
+using SFA.DAS.ProviderCommitments.Interfaces;
+
+namespace SFA.DAS.ProviderCommitments.Queries.GetTrainingCourses
+{
+    public sealed class GetTrainingCoursesQueryHandler : IRequestHandler<GetTrainingCoursesQueryRequest, GetTrainingCoursesQueryResponse>
+    {
+        private readonly IApprenticeshipInfoService _apprenticeshipInfoService;
+
+        public GetTrainingCoursesQueryHandler(IApprenticeshipInfoService apprenticeshipInfoService)
+        {
+            _apprenticeshipInfoService = apprenticeshipInfoService;
+        }
+
+        public async Task<GetTrainingCoursesQueryResponse> Handle(GetTrainingCoursesQueryRequest message, CancellationToken cancellationToken)
```

Comment:
I'm not feeling very much love for this method. I can't tell immediately what it is doing. Could we restructure it to make it a bit more comprehensible? 

The cancellation token should be recognised and we're awaiting standardsTask on both branches of the if statement. The nested awaits look a bit clumsy.

Reply:
Addressed by https://github.com/SkillsFundingAgency/das-providercommitments/pull/8/commits/12381b3334d75ec10e71f682cad98ef85a441ff5